include/boost/geometry/core/mutable_range.hpp

```diff
@@ -75,6 +76,31 @@ struct push_back
     {
         range.push_back(item);
     }
+
+    static inline void apply(typename rvalue_type<Range>::type range,
+                 item_type && item)
+    {
+        range.push_back(std::move(item));
+    }
+};
+
+
+/*!
+\brief Traits class to append an element of geometry collection
+\ingroup mutable_range
+ */
+template <typename Range>
+struct emplace_back
+{
+    // When specializing it'd be enough to only implement it for one argument
+    // because we'll use it to pass one object of type potentially different
+    // than range_value but which range_value can be constructed from.
+    template <typename ...Args>
+    static inline void apply(typename rvalue_type<Range>::type range,
+                             Args&&... args)
+    {
+        range.emplace_back(std::forward<Args>(args)...);
+    }
```

Comment:
This will break older code? (with own types)? Or is it optional?

Reply:
Not by itself. It could break the code if we replaced `push_back` with it and a user-defined geometry wouldn't also define `emplace_back`.

This is why `emplace_back` should be used only with GCs. Why not `push_back`? Because a GC can store a different type than the one that is passed. E.g. variant is stored but we're emplacing back a point, or GC stores pointers to geometries but we pass a linestring.

For all other geometries we know the type that is stored and we should keep using `push_back`. However now we can also move which will perform copy if the user-defined geometry doesn't define move ctor.